### PR TITLE
New hemophage quirk

### PR DESCRIPTION
**Pull request by plsleavemealon**
## About The Pull Request
A new quirk added for hemophages, lets them eat or drink normal food.

They gain no nutritional benefit from it, but now they can taste what they are eating without immediately vomiting, thus letting them roleplay with others when they make fancy foods or drinks.

I added in some new flags to the hemophage liver/stomach (corrupted liver/stomach) which functions identical to the ageusia quirk, in that it blocks them from gaining disgust from eating normal non-bloody food, however, it doesn't block them from tasting it!

## Why It's Good For The Game
This is a largely roleplay oriented quirk that allows hemophages to enjoy food alongside their friends and be able to describe what they are tasting to them.  Costs 2 points, since its an "improved" form of ageusia for them.

If I knew how, I would also include the ability for hemophages to pick preferred food types (and lock in bloody as one of the favorites), letting them customize their reaction to various foods.

## Proof Of Testing
Tested it on my private server and it works flawlessly as far as I can tell

## Changelog

:cl:
add: New Quirk: Masquerade, exclusive to hemophages
/:cl:

### CODE REVIEW

File: code/__DEFINES/traits/declarations.dm [hunk 1]
```diff
@@ -803,6 +803,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_THROWINGARM "throwing_arm"
 #define TRAIT_SETTLER "settler"
 #define TRAIT_STRONG_STOMACH "strong_stomach"
+#define TRAIT_MASQUERADE_FOOD "masquerade_food"
```
Comment (StrangeWeirdKitten) [inline code/__DEFINES/traits/declarations.dm:806]: Put this in the ~~bubber __DEFINES folder.

File: code/_globalvars/traits/admin_tooling.dm [hunk 2]
```diff
@@ -120,6 +120,7 @@ GLOBAL_LIST_INIT(admin_visible_traits, list(
 		"TRAIT_LITERATE" = TRAIT_LITERATE,
 		"TRAIT_LIVERLESS_METABOLISM" = TRAIT_LIVERLESS_METABOLISM,
 		"TRAIT_MAGICALLY_GIFTED" = TRAIT_MAGICALLY_GIFTED,
+		"TRAIT_MASQUERADE_FOOD" = TRAIT_MASQUERADE_FOOD,
```
Comment (StrangeWeirdKitten) [inline code/_globalvars/traits/admin_tooling.dm:123]:
```suggestion
		"TRAIT_MASQUERADE_FOOD" = TRAIT_MASQUERADE_FOOD, //BUBBER EDIT
```

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage/corrupted_liver.dm [hunk 3]
```diff
@@ -38,6 +38,9 @@
 	if(HAS_TRAIT(target_mob, TRAIT_AGEUSIA)) // They don't taste anything, their body shouldn't react strongly to the taste of that stuff.
 		return
 
+	if(HAS_TRAIT(target_mob, TRAIT_MASQUERADE_FOOD)) // Their tumor has adapted the ability to consume normal food without violently rejecting it.
+		return
+
```
Comment (StrangeWeirdKitten) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage/corrupted_liver.dm:43]:
```suggestion
// BUBBER EDIT START
	if(HAS_TRAIT(target_mob, TRAIT_MASQUERADE_FOOD)) // Their tumor has adapted the ability to consume normal food without violently rejecting it.
		return
// BUBBER EDIT END
```

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage/corrupted_stomach.dm [hunk 4]
```diff
@@ -34,6 +34,9 @@
 	if(parent_organ.owner && HAS_TRAIT(parent_organ.owner, TRAIT_AGEUSIA)) // They don't taste anything, their body shouldn't react strongly to the taste of that stuff.
 		return
 
+	if(parent_organ.owner && HAS_TRAIT(parent_organ.owner, TRAIT_MASQUERADE_FOOD)) // Their tumor has adapted the ability to consume normal food without violently rejecting it.
+		return
+
```
Comment (StrangeWeirdKitten) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage/corrupted_stomach.dm:39]:
```suggestion
// BUBBER EDIT START
	if(parent_organ.owner && HAS_TRAIT(parent_organ.owner, TRAIT_MASQUERADE_FOOD)) // Their tumor has adapted the ability to consume normal food without violently rejecting it.
		return
// BUBBER EDIT END
```

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]

**Comment by StrangeWeirdKitten** [comment]
Also shrink the changelog. You only need a single Add:

File: code/__DEFINES/traits/declarations.dm (hunk 1 above)
Comment (plsleavemealon) [reply code/__DEFINES/traits/declarations.dm:806]: Which one is that?

Comment (plsleavemealon) [reply code/__DEFINES/traits/declarations.dm]: NVM, found it I think!

File: code/_globalvars/traits/_traits.dm [hunk 5]
```diff
@@ -307,6 +307,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_MAGICALLY_PHASED" = TRAIT_MAGICALLY_PHASED,
 		"TRAIT_MARTIAL_ARTS_IMMUNE" = TRAIT_MARTIAL_ARTS_IMMUNE,
 		"TRAIT_MANSUS_TOUCHED" = TRAIT_MANSUS_TOUCHED,
+		"TRAIT_MASQUERADE_FOOD" = TRAIT_MASQUERADE_FOOD,
```
Comment (StrangeWeirdKitten) [inline code/_globalvars/traits/_traits.dm:310]:
```suggestion
		"TRAIT_MASQUERADE_FOOD" = TRAIT_MASQUERADE_FOOD, // BUBBER EDIT
```

**Review by StrangeWeirdKitten — APPROVED** [review]

**Pull request merged**